File: python/src/generate_intent_identifier_example.py
from libra import identifier, utils
from libra.libra_types import AccountAddress


def generate_intent_identifier(address: AccountAddress, amount: int, currency: str) -> str:
    account_identifier = identifier.encode_account(utils.account_address_hex(address), None, identifier.TLB)
    intent_identifier = identifier.encode_intent(account_identifier, currency, amount)
    print(f"intent_identifier: {intent_identifier}")

    return intent_identifier

File: java/src/main/java/example/GenerateIntentIdentifierExample.java
package example;

import org.libra.AccountIdentifier;
import org.libra.IntentIdentifier;
import org.libra.types.AccountAddress;

import static org.libra.AccountIdentifier.NetworkPrefix.TestnetPrefix;

public class GenerateIntentIdentifierExample {
    public static String generateIntentIdentifier(AccountAddress address, long amount, String currency) {
        AccountIdentifier accountIdentifier = new AccountIdentifier(TestnetPrefix, address);
        IntentIdentifier intentIdentifier = new IntentIdentifier(accountIdentifier, currency, amount);
        String encodedIntentIdentifier = intentIdentifier.encode();

        System.out.println("encodedIntentIdentifier: " + encodedIntentIdentifier);

        return encodedIntentIdentifier;
    }
}
